src/utils/send-payments.ts

```diff
@@ -73,27 +73,30 @@ export async function sendSignedTransactions(
   let continueSending = true;
   let timeout = 5000;
   let nonce = await getNonce(keys.publicKey);
-  nonce++;
   
   if (totalBurn > 0) {
-      const burnTransaction: payment = { to: burnAddress, from: keys.publicKey, fee: payorSendTransactionFee, amount: totalBurn, nonce: nonce, memo: memo };
-      const burnWorked = await sendSignedBurnTransactions(burnTransaction, keys);
-    nonce = burnWorked ? nonce++ : nonce;
+    const burnTransaction: payment = { to: burnAddress, from: keys.publicKey, fee: payorSendTransactionFee, amount: totalBurn, nonce: nonce, memo: memo };
+    const burnWorked = await sendSignedBurnTransactions(burnTransaction, keys);
+    if (burnWorked) {
+      nonce = nonce++;
```

Comment:
nonce should increment anyway if continueSending = false; the app will continue processing all the other keys to generate, sign, and log their transactions to a nonce file, so that the resender can send them later. 

This should either a) abort processing altogether, or b) increment the nonce and let the app continue logging (but not sending) the other payments.

While b seems like the obvious choice, a may be better. This was implemented because of issues with transactions getting dropped; it would be rare for that to be the problem on the very first transaction send of the run, so it may indicate that there is something else wrong and aborting might be a better behavior. I'm on the fence.

Reply:
Fixed in the latest commit to increment the nonce despite the failed burn